.github/actions/setup-opensearch-dashboards/action.yml

```diff
@@ -0,0 +1,269 @@
+    value: ${{ steps.determine-dashboards-directory-zip-install.outputs.dashboards-directory }}
+
+  plugin-directory:
+    description: "The directory where the plugin has been configured"
+    value: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
+
+
+runs:
+  using: "composite"
+  steps:
+    - id: determine-dashboards-directory
+      if: ${{ inputs.install_zip != 'true' }}
+      run: echo "dashboards-directory=OpenSearch-Dashboards"  >> $GITHUB_OUTPUT
+      shell: bash
+
+    - id: determine-plugin-directory
+      run: echo "::set-output name=plugin-directory::./OpenSearch-Dashboards/plugins/${{ inputs.plugin_name }}"
+      shell: bash
+
+    - uses: actions/checkout@v2
+      with:
+        path: OpenSearch-Dashboards
+        repository: cwperks/OpenSearch-Dashboards
+        ref: 'support-repeated-plugin'
+        fetch-depth: 0
+        filter: |
+          cypress
+          test
+
+    - uses: actions/checkout@v2
+      with:
+        path: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
+
+    - uses: actions/checkout@v2
+      with:
+        path: ./OpenSearch-Dashboards/plugins/security-admin-dashboards-plugin
+
+    - id: osd-version
+      run: |
+        echo "::set-output name=osd-version::$(jq -r '.opensearchDashboards.version | split(".") | .[:2] | join(".")' package.json)"
+        echo "::set-output name=osd-x-version::$(jq -r '.opensearchDashboards.version | split(".") | .[0]' package.json).x"
+        echo "::set-output name=plugin-version::$(jq -r '.opensearchDashboardsVersion' opensearch_dashboards.json)"
+      working-directory: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
+      shell: bash
+
+    # - id: branch-switch-if-possible
+    #   continue-on-error: true # Defaults onto main if the branch switch doesn't work
+    #   if: ${{ steps.osd-version.outputs.osd-version }}
+    #   run: git checkout ${{ steps.osd-version.outputs.osd-version }} || git checkout ${{ steps.osd-version.outputs.osd-x-version }}
+    #   working-directory: ./OpenSearch-Dashboards
+    #   shell: bash
+
+    - id: tool-versions
+      run: |
+        echo "node_version=$(cat .node-version)" >> $GITHUB_OUTPUT
+        echo "yarn_version=$(jq -r '.engines.yarn' package.json)" >> $GITHUB_OUTPUT
+      working-directory: OpenSearch-Dashboards
+      shell: bash
+
+    - uses: actions/setup-node@v1
+      with:
+        node-version: ${{ steps.tool-versions.outputs.node_version }}
+        registry-url: 'https://registry.npmjs.org'
+
+    - name: Setup Opensearch Dashboards
+      run: |
+        npm uninstall -g yarn
+        echo "Installing yarn ${{ steps.tool-versions.outputs.yarn_version }}"
+        npm i -g yarn@${{ steps.tool-versions.outputs.yarn_version }}
+        yarn cache clean
+        yarn add sha.js
+      working-directory: OpenSearch-Dashboards
+      shell: bash
+
+    - name: Create opensearch_dashboards.json for security admin dashboards plugin
```

Comment:
I suggest we add a `working-directory` for this at `./OpenSearch-Dashboards/plugins/security-admin-dashboards-plugin`